### PR TITLE
(embassy-boot): add blocking API to FirmwareUpdater

### PR DESCRIPTION
**Pull request by MathiasKoch**
Also add a split `prepare_update` + `write_firmware` API, to allow for an optimized update API at the exchange of added complexity. 
The old API is left in place to allow users to choose the complexity level that fits their needs.

### CODE REVIEW

File: embassy-boot/boot/src/lib.rs [hunk 1]
```diff
@@ -679,7 +688,156 @@ impl FirmwareUpdater {
...
+
+        Ok(())
+    }
+
+    /// Prepare for an incoming DFU update by erasing the entire DFU area and
+    /// returning a `FirmwareWriter`.
+    ///
+    /// Using this instead of `write_firmware` allows for an optimized API in
+    /// exchange for added complexity.
+    pub fn prepare_update<F: NorFlash>(&mut self, flash: &mut F) -> Result<FirmwareWriter, F::Error> {
```
Comment (lulf) [inline embassy-boot/boot/src/lib.rs:703]: From what I can tell, this is identical to prepare_update_blocking, so it should perhaps instead be async and take the async NorFlash trait instead?

**Review by lulf — COMMENTED** [review]
Looks good, I think one minor change is needed for prepare_update to be async.

File: embassy-boot/boot/src/lib.rs (hunk 1 above)
Comment (MathiasKoch) [reply embassy-boot/boot/src/lib.rs:703]: Huh, nice catch. A bit too fast on the copy-paste :)

**Comment by MathiasKoch** [comment]
@lulf Fixed :+1:

**Comment by lulf** [comment]
Thanks, great work :rocket: 

bors r+

**Pull request merged**